src/storage/txn/actions/acquire_pessimistic_lock.rs

```diff
@@ -140,8 +142,9 @@ pub fn acquire_pessimistic_lock<S: Snapshot>(
     }
 
     // Following seek_write read the previous write.
+    let mut key_exists = false;
```

Comment:
I think this one can be just replaced with `val.is_some()`

Reply:
done